src/parameterized/LKJL.jl

```diff
@@ -38,17 +50,24 @@ function logdensity(d::LKJL{k}, L::Union{LinearAlgebra.AbstractTriangular, Diago
     η = d.η
     # z = diag(L)
     # sum(log.(z) .* ((k:-1:1) .+ 2*(η-1)))
-    c = k + 1 + 2(η - 1)
+
+    # Note: https://github.com/cscherrer/MeasureTheory.jl/issues/100#issuecomment-852428192
+    c = k + 2(η - 1)
     @tullio s = (c - i) * log(L[i,i])
     return s
 end
 
 using TransformVariables
 
-TransformVariables.as(::LKJL{k}) where {k} = TransformVariables.CorrCholeskyFactor(k)
+TransformVariables.as(::LKJL{k}) where {k} = CorrCholeskyLower(k)
 
+# TODO: I think this is wrong
 function basemeasure(μ::LKJL{k}) where {k}
     t = as(μ)
     d = dimension(t)
     return Pushforward(t, Lebesgue(ℝ)^d, false)
 end
+
+function Base.rand(rng::AbstractRNG, ::Type, d::LKJL{k}) where {k}
```

Comment:
Note: this can be done without the `cholesky` decomposition, by randomly sampling the canonical partial correlations, as in the LKJ paper, and then mapping them to the cholesky factor instead of the correlation matrix. Stan [implements this](https://github.com/stan-dev/math/blob/develop/stan/math/prim/prob/lkj_corr_cholesky_rng.hpp). But that can be for a future PR.

The `::Type` parameter is not used. Should it be? Also, does it matter for MeasureTheory if the output type is `LowerTriangular` or a lower triangular `Matrix`?

Reply:
I'm not entirely happy with our random generation, so I'm open to ideas in general. But this `::Type` setup is one aspect I think is very nice. 

This comes from @devmotion in https://github.com/cscherrer/MeasureTheory.jl/issues/51; the idea is that every `rand` call will ententually (after a series of dispatches) call at least one `rand(::AbstractRNG, T::Type)`. The resulting `::T` will the propagate back up to the final result.

Reply:
> Also, does it matter for MeasureTheory if the output type is LowerTriangular or a lower triangular Matrix?

I'm not sure. I haven't dug into the internals of `LowerTriangular`, but here's one thing that's pretty interesting:
```julia
julia> X = randn(4,4)
4×4 Matrix{Float64}:
  0.402848  0.787592  -0.110378   0.669417
  0.183548  0.48167    0.8996    -0.616217
 -0.108725  1.62248   -1.0993    -0.143815
 -0.305961  0.341102   1.95539    0.849061

julia> sizeof(X)
128

julia> using LinearAlgebra

julia> sizeof(LowerTriangular(X))
8
```

This seems like a big advantage, but strangely I don't see a way to construct one directly (only to build a Matrix and then convert it). What would you suggest?

Reply:
> The resulting ::T will the propagate back up to the final result.

I see, so the reason `::Type` is unused here is because Distributions's `rand` function doesn't accept this type parameter, correct? Okay, what was confusing me was that it was there but unused.

> I'm not sure. I haven't dug into the internals of `LowerTriangular`, but here's one thing that's pretty interesting:

Weird. I have no idea what `sizeof` does, and it just redirects to a `Core.sizeof` function, which `@which` does not work for, saying its not a generic function, so I don't know what this even means.

`LowerTriangular` has a simple implementation though. It's just a wrapper around another matrix for storage, where it only considers the lower triangle of that matrix. So in your example, `LowerTriangular` just wraps your original `X`; it is no more or less space-efficient than `X`. Potentially by wrapping you could leave the strict upper triangle with undefined entries, but not even functions in LinearAlgebra seem to do that.

Reply:
> I see, so the reason `::Type` is unused here is because Distributions's `rand` function doesn't accept this type parameter, correct?

Yep, that's it. David had a PR to add this to Distributions, but I'm not sure of the status.

> Weird. I have no idea what sizeof does, and it just redirects to a Core.sizeof function, which @which does not work for, saying its not a generic function, so I don't know what this even means.

It's suposed to be the size in bytes. But looking again I think there must be a bug for this case.

> LowerTriangular has a simple implementation though. It's just a wrapper around another matrix for storage, where it only considers the lower triangle of that matrix. So in your example, LowerTriangular just wraps your original X; it is no more or less space-efficient than X. Potentially by wrapping you could leave the strict upper triangle with undefined entries, but not even functions in LinearAlgebra seem to do that.

Ah, ok. In that case I think we do want `LowerTriangular`. We can allocate the matrix as `undef`, write only the values we need, and the constructor will "mask" the remaining `undef`s